nemo/deploy/nlp/trtllm_pytorch_deployable.py

```diff
@@ -0,0 +1,209 @@
+        """
+        while True:
+            message = torch.empty(1, dtype=torch.long, device="cuda")
+            torch.distributed.broadcast(message, src=0)
+            if message == 0:
+                prompts = broadcast_list(data=[None], src=0)
+                temperature, top_k, top_p, max_length = broadcast_list(data=[None], src=0)
+
+                self.generate(
+                    prompts=prompts,
+                    temperature=temperature,
+                    top_k=top_k,
+                    top_p=top_p,
+                    max_length=max_length,
+                )
+            else:
+                return
+
+    @property
+    def get_triton_input(self):
+        inputs = (
+            Tensor(name="prompts", shape=(-1,), dtype=bytes),
+            Tensor(name="max_length", shape=(-1,), dtype=np.int_, optional=True),
+            Tensor(name="max_batch_size", shape=(-1,), dtype=np.int_, optional=True),
+            Tensor(name="top_k", shape=(-1,), dtype=np.int_, optional=True),
+            Tensor(name="top_p", shape=(-1,), dtype=np.single, optional=True),
+            Tensor(name="temperature", shape=(-1,), dtype=np.single, optional=True),
+            Tensor(name="random_seed", shape=(-1,), dtype=np.int_, optional=True),
+        )
+        return inputs
+
+    @property
+    def get_triton_output(self):
+        return (Tensor(name="sentences", shape=(-1,), dtype=bytes),)
+
+    @batch
+    @first_value("temperature", "top_k", "top_p", "max_length")
+    def triton_infer_fn(self, **inputs: np.ndarray):
+        output_infer = {}
+
+        try:
+            prompts = str_ndarray2list(inputs.pop("prompts"))
+            temperature = inputs.pop("temperature", None)
+            top_k = int(inputs.pop("top_k")) if "top_k" in inputs else None
+            top_p = inputs.pop("top_p", None)
+            max_length = inputs.pop("max_length", 256)
+
+            if torch.distributed.is_initialized():
+                if torch.distributed.get_world_size() > 1:
+                    torch.distributed.broadcast(torch.tensor([0], dtype=torch.long, device="cuda"), src=0)
+                    broadcast_list(prompts, src=0)
+                    broadcast_list(
+                        data=[
+                            temperature,
+                            top_k,
+                            top_p,
+                            max_length,
+                        ],
+                        src=0,
+                    )
+
+            output = self.generate(
+                prompts=prompts,
+                temperature=temperature,
+                top_k=top_k,
+                top_p=top_p,
+                max_length=max_length,
+            )
+
+            output_infer = {"sentences": cast_output(output, np.bytes_)}
+
+        except Exception as error:
+            err_msg = "An error occurred: {0}".format(str(error))
+            output_infer["sentences"] = cast_output([err_msg], np.bytes_)
```

Comment:
This error handling is buggy. If the `try` block errors out and `len(prompts) > 1`, then Pytriton will complain about wrong-sized output: `1` vs `len(prompts)`. This clutters real issues a lot, making debugging really hard.

Suggestion:
* **recommended**: let's get rid of try / except blocks here: its intent is not clear anyway
* WAR: let's make sure to return correctly sized "sentences" in the `except` block (see [here](https://github.com/NVIDIA/NeMo/blob/9cc90fc7f44ea44738ee1ece1aee89458f4d70e8/nemo/export/tensorrt_llm.py#L1591) from https://github.com/NVIDIA/NeMo/pull/12773)